software/statemachine.py

```diff
@@ -0,0 +1,278 @@
+from enum import Enum
+import math
+
+class State(Enum):
+    FIND_BALL = 1
+    GO_TO_BALL = 2
+    ORBIT = 3
+    THROW = 4
+    TESTING = 5
+    WAIT_REFEREE = 6
+    GO_TO_BASKET = 7
+
+class StateMachine:
+    def __init__(self, omniRobot):
+        self.currentState = State.FIND_BALL
+        self.robot = omniRobot
+        self.throwIntoBlue = False
+
+        self.imageData = None
+        self.imageWidth = 848
```

Comment:
Should the camera not define this? Else if you change it, this part of the logic will break.

Reply:
Removed hardcoded values.